CodeEditor/Sources/CodeEditor/UXCodeTextViewRepresentable.swift

```diff
@@ -202,31 +202,38 @@ public struct UXCodeTextViewRepresentable: UXViewRepresentable {
         if editorBindings.themeName.rawValue != textView.themeName.rawValue {
             textView.applyNewTheme(editorBindings.themeName)
         }
-        textView.language = editorBindings.language
         
+        textView.language = editorBindings.language
         textView.indentStyle          = editorBindings.indentStyle
         textView.isSmartIndentEnabled = editorBindings.flags.contains(.smartIndent)
         
+        var deletedHighlights = textView.highlightedRanges.filter { !editorBindings.highlightedRanges.contains($0)
+        }
+        
         if editorBindings.source.wrappedValue != textView.string {
             // reset contentoffset when switching files as it is not stored and content heights of files vary
+            deletedHighlights.removeAll()
             textView.setContentOffset(CGPoint(x: 0, y: 0), animated: false)
             if let textStorage = textView.codeTextStorage {
-                textStorage.replaceCharacters(in: NSRange(location: 0, length: textStorage.length),
-                                              with: editorBindings.source.wrappedValue)
+                textStorage.replaceCharacters(
+                    in: NSRange(location: 0, length: textStorage.length),
+                    with: editorBindings.source.wrappedValue)
             } else {
                 assertionFailure("no text storage?")
+                print("test")
```

Comment:
Do we really need this?

Reply:
Oops, doesn't get called anyway but will remove it